packages/client/hmi-client/src/components/model/petrinet/tera-parameter-table-v2.vue

```diff
@@ -49,41 +99,99 @@
+import { getParameterDistribution, getParameters } from '@/services/model-configurations';
 import { StratifiedMatrix } from '@/types/Model';
-import { ref } from 'vue';
+import { computed, ref } from 'vue';
 import { collapseParameters, isStratifiedModel } from '@/model-representation/mira/mira';
 import { MiraModel, MiraTemplateParams } from '@/model-representation/mira/mira-common';
 import Accordion from 'primevue/accordion';
 import AccordionTab from 'primevue/accordiontab';
 import Button from 'primevue/button';
 import Divider from 'primevue/divider';
-import { DistributionType } from '@/services/distribution';
-import TeraStratifiedMatrixModal from './model-configurations/tera-stratified-matrix-modal.vue';
+import { DistributionType, distributionTypeOptions } from '@/services/distribution';
+import InputNumber from 'primevue/inputnumber';
+import Dropdown from 'primevue/dropdown';
+import Checkbox from 'primevue/checkbox';
 import TeraParameterEntry from './tera-parameter-entry.vue';
+import TeraStratifiedMatrixModal from './model-configurations/tera-stratified-matrix-modal.vue';
 
 const props = defineProps<{
 	modelConfiguration: ModelConfiguration;
 	mmt: MiraModel;
 	mmtParams: MiraTemplateParams;
 }>();
 
-const emit = defineEmits(['update-parameter', 'update-source', 'delete-distribution']);
+const emit = defineEmits(['update-parameters', 'update-source']);
 
 const isStratified = isStratifiedModel(props.mmt);
 
+const isAddingUncertainty = ref(false);
+const uncertaintyType = ref(DistributionType.Uniform);
+const uncertaintyPercentage = ref(10);
+const selectedParameters = ref<string[]>([]);
+
+const numParameters = computed(() => Object.keys(props.mmt.parameters).length);
+
 const matrixModalId = ref('');
+
+const onAddUncertainty = () => {
+	const selected = Object.keys(props.mmt.parameters);
+	selectedParameters.value = selected;
+	isAddingUncertainty.value = true;
+};
+
+const onSelect = (paramId: string) => {
+	console.log(paramId);
+	if (selectedParameters.value.includes(paramId)) {
+		selectedParameters.value.splice(selectedParameters.value.indexOf(paramId), 1);
+	} else {
+		selectedParameters.value.push(paramId);
+	}
+};
+
+const onUpdateDistributions = () => {
+	const distributions: { id: string; distribution: ModelDistribution }[] = [];
+	if (uncertaintyType.value === DistributionType.Uniform) {
+		selectedParameters.value.forEach((paramId) => {
+			const parameter = getParameterDistribution(props.modelConfiguration, paramId);
+			if (parameter.type !== DistributionType.Constant) return;
+			const distribution = {
+				id: paramId,
+				distribution: {
+					type: uncertaintyType.value,
+					parameters: {
+						minimum:
+							parameter.parameters.value -
+							(parameter.parameters.value * uncertaintyPercentage.value) / 100,
+						maximum:
+							parameter.parameters.value +
+							(parameter.parameters.value * uncertaintyPercentage.value) / 100
```

Comment:
Some comments on the `onUpdateDistributions` function:

- The naming convention here is confusing and can make the logic harder to follow, for example `distributions` and `distribution` are different types of objects, where one would expect them to be the same. On top of that "distributions" get feed into "parameters" slot in setParameterDistributions ... 🤯 

- I think the math parts are more readable if we don't repeat so much

```
const v = parameter.parameters.value;
const delta = (parameter.parameters.value * uncertaintyPercentage.value) / 100;
...
minimum: v - delta,
maximum: v + delta
```

- More of a nitpick and I am not totally sure, but I think in most place we specify percentage between `[0-1]` rather than `[0-100]`.

Reply:
I'll make some changes in the morning 😁

Reply:
I've renamed everything so that its clearer (i hope) and fixed up the math

Reply:
cool thx